frontend/src/features/statistics/StatsContext.tsx

```diff
@@ -0,0 +1,91 @@
+import React, {
+  createContext,
+  useCallback,
+  useMemo,
+  useRef,
+  useState,
+} from 'react';
+import { Statistics, StatsService } from 'src/services/openapi';
+
+// Stats are considered outdated after this amount of miliseconds.
+const EXPIRATION_TIME = 4000;
+
+interface StatsContextValue {
+  stats: Statistics;
+  getStats: () => Statistics;
+  refreshStats: () => void;
+}
+
+const initialState: Statistics = {
+  active_users: {
+    total: 0,
+    joined_last_month: 0,
+    joined_last_30_days: 0,
+  },
+  polls: [],
+};
+
+const StatsContext = createContext<StatsContextValue>({
+  stats: initialState,
+  getStats: () => initialState,
+  refreshStats: () => undefined,
+});
+
+export const StatsLazyProvider = ({
+  children,
+}: {
+  children: React.ReactNode;
+}) => {
+  const loading = useRef(false);
+  const lastRefreshAt = useRef(0);
+
+  const [stats, setStats] = useState(initialState);
+
+  const refreshStats = useCallback(async () => {
+    setStats(await StatsService.statsRetrieve());
+    loading.current = false;
+    lastRefreshAt.current = Date.now();
+  }, []);
+
+  /**
+   * Return the current `stats` if any, else refresh them.
+   */
+  const getStats = useCallback(() => {
+    const currentTime = Date.now();
+
+    // Initialize the stats if they are empty.
+    if (stats.polls.length === 0 && !loading.current) {
+      loading.current = true;
+      refreshStats();
+      return initialState;
+    }
+
+    // Refresh the stats when the existing stats are outdated.
+    if (
+      stats.polls.length !== 0 &&
+      currentTime - lastRefreshAt.current >= EXPIRATION_TIME
+    ) {
+      refreshStats();
+      return stats;
+    }
+
+    return stats;
+  }, [stats, refreshStats]);
```

Comment:
`loading` is not used when stats are outdated, so concurrent requests can be triggered in this case.
The refresh should never be called when another request is pending:

```js
  const getStats = useCallback(() => {
    const currentTime = Date.now();

    if (loading.current) {
      return stats;
    }

    // Initialize the stats if they are empty, or if they are outdated.
    if (
      stats.polls.length === 0 ||
      currentTime - lastRefreshAt.current >= EXPIRATION_TIME
    ) {
      loading.current = true;
      refreshStats();
    }
    return stats;
  }, [stats, refreshStats]);

Reply:
Should be fixed by https://github.com/tournesol-app/tournesol/pull/1542/commits/f1e37f3fcf7aad563122847088fd7d9064a444b6